src/algebraic_topology/simplex_category.lean

```diff
@@ -0,0 +1,252 @@
+lemma δ_comp_δ_self {n} {i : fin (n+2)} : δ i ≫ δ i.cast_succ = δ i ≫ δ i.succ :=
+begin
+  ext j,
+  dsimp [δ, fin.succ_above],
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  split_ifs; { simp at *; linarith },
+end
+
+/-- The second simplicial identity -/
+lemma δ_comp_σ_of_le {n} {i : fin (n+2)} {j : fin (n+1)} (H : i ≤ j.cast_succ) :
+  δ i.cast_succ ≫ σ j.succ = σ j ≫ δ i :=
+begin
+  ext k,
+  dsimp [δ, σ, fin.succ_above, fin.pred_above],
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  rcases k with ⟨k, _⟩,
+  simp only [subtype.mk_le_mk, fin.cast_succ_mk] at H,
+  simp with push_cast, -- `simp?` doesn't work here
+  split_ifs,
+  -- Hope for the best from `linarith`:
+  all_goals { simp at *, try { linarith }, },
+  -- Two of the goals need special handling:
+  { replace h_3 := nat.le_of_pred_lt h_3, change k ≤ i at h_3, linarith, },
+  { exact (nat.succ_pred_eq_of_pos (lt_of_le_of_lt (zero_le _) h_1)).symm, },
+end
+
+/-- The first part of the third simplicial identity -/
+lemma δ_comp_σ_self {n} {i : fin (n+1)} :
+  δ i.cast_succ ≫ σ i = 𝟙 _ :=
+begin
+  ext j,
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  dsimp [δ, σ, fin.succ_above, fin.pred_above],
+  simp with push_cast,
+  split_ifs; { simp at *; linarith, },
+end
+
+/-- The second part of the third simplicial identity -/
+lemma δ_comp_σ_succ {n} {i : fin (n+1)} :
+  δ i.succ ≫ σ i = 𝟙 _ :=
+begin
+  ext j,
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  dsimp [δ, σ, fin.succ_above, fin.pred_above],
+  simp with push_cast,
+  split_ifs; { simp at *; linarith, },
+end
+
+/-- The fourth simplicial identity -/
+lemma δ_comp_σ_of_gt {n} {i : fin (n+2)} {j : fin (n+1)} (H : j.cast_succ < i) :
+  δ i.succ ≫ σ j.cast_succ = σ j ≫ δ i :=
+begin
+  ext k,
+  dsimp [δ, σ, fin.succ_above, fin.pred_above],
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  rcases k with ⟨k, _⟩,
+  simp at H,
+  -- rw apply_dite fin.cast_succ,
+  simp [apply_dite fin.cast_succ] with push_cast, -- `simp?` doesn't work here
+  split_ifs,
+  -- Hope for the best from `linarith`:
+  any_goals { simp at *, try { linarith }, },
+  -- Four of the goals need special handling:
+  { simp at h_1,
+    exact lt_irrefl (k - 1) (lt_of_lt_of_le
+     (nat.pred_lt (ne_of_lt (lt_of_le_of_lt (zero_le _) h_1)).symm)
+     (le_trans (nat.le_of_lt_succ h) h_2)) },
+  { simp at h_1, linarith, },
+  { exfalso, exact lt_irrefl _ (lt_of_le_of_lt (nat.le_pred_of_lt (nat.lt_of_succ_le h)) h_3), },
+  { exact (nat.succ_pred_eq_of_pos (lt_of_le_of_lt (zero_le _) h_2)).symm, },
+end
+
+local attribute [simp] fin.pred_mk
+
+/-- The fifth simplicial identity -/
+lemma σ_comp_σ {n} {i j : fin (n+1)} (H : i ≤ j) :
+  σ i.cast_succ ≫ σ j = σ j.succ ≫ σ i :=
+begin
+  ext k,
+  dsimp [σ, fin.pred_above],
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  rcases k with ⟨k, _⟩,
+  simp at H,
+  simp with push_cast,
+  split_ifs,
+  -- Hope for the best from `linarith`:
+  any_goals { simp at *, try { linarith }, },
```

Comment:
Non-terminal simp.

Reply:
I've rearranged this, dealing with the two "special cases" first, so that this `simp at *; linarith` becomes terminal.